core/src/main/java/com/airbnb/aerosolve/core/features/Features.java

```diff
@@ -73,31 +73,43 @@ public Example toExample(boolean isMultiClass) {
     stringFeatures.put("BIAS", bias);
     stringFeatures.put(MISS, missing);
 
+    // metadata map
+    final Map<String, String> metadata = new HashMap<>();
+    example.setMetadata(metadata);
+
     for (int i = 0; i < names.length; i++) {
       String name = names[i];
       Object value = values[i];
-      if (value == null) {
-        missing.add(name);
+      if (isMetadata(name)) {
+        metadata.put(name.substring(1), value == null ? null : value.toString());
       } else {
-        Pair<String, String> feature = getFamily(name);
-        if (value instanceof String) {
-          String str = (String) value;
-          if (isMultiClass && isLabel(feature)) {
-            addMultiClassLabel(str, floatFeatures);
+        if (value == null) {
+          missing.add(name);
+        } else {
+          Pair<String, String> feature = getFamily(name);
+          if (value instanceof String) {
+            String str = (String) value;
+            if (isMultiClass && isLabel(feature)) {
+              addMultiClassLabel(str, floatFeatures);
+            } else {
+              addStringFeature(str, feature, stringFeatures);
+            }
+          } else if (value instanceof Boolean) {
+            Boolean b = (Boolean) value;
+            addBoolFeature(b, feature, stringFeatures);
           } else {
-            addStringFeature(str, feature, stringFeatures);
+            addNumberFeature((Number) value, feature, floatFeatures);
           }
-        } else if (value instanceof Boolean) {
-          Boolean b = (Boolean) value;
-          addBoolFeature(b, feature, stringFeatures);
-        } else {
-          addNumberFeature((Number) value, feature, floatFeatures);
         }
       }
     }
     return example;
   }
 
+  private static boolean isMetadata(String name) {
+    return name.charAt(0) == FAMILY_SEPARATOR;
```

Comment:
`#`? Not sure whether this or other characters are good choices for Hive column names though.